### DISCUSSION TITLE
Log user, token info in access.log

### DISCUSSION THREAD

**Discussion by bodtx** [OpenIDC/mod_auth_openidc#965] [Q&A] [answered]
Hello, apache can output user logged in with basic auth with %u in LogFormat:

`LogFormat "%h %l  %t %u  \"%r\" %>s %b \"%{Referer}i\" \"%{User-agent}i\"" userLog`

here %u is displaying ([sub]@[iss]) 

Is there a way to display others info from JWT token (mail, roles)?
(BTW I have no PHP mode enabled)

thx

**Comment by zandbelt** [comment, marked as answer]
you should be able to reference the environment variables set by mod_auth_openidc by using e.g. `%{OIDC_CLAIM_email}e`

**Comment by hmoffatt** [comment]
On 26/11/22 02:00, bodtx wrote:
>
> Hello, apache can output user logged in with basic auth with %u in 
> LogFormat:
>
> |LogFormat "%h %l %t %u \"%r\" %>s %b \"%{Referer}i\" 
> \"%{User-agent}i\"" userLog|
>
> here %u is not working
>
> Is there a way to display the connected user or show some info from 
> JWT token (mail, roles)?
> (BTW I have no PHP mode enabled)
>

I have the following in my config, and I get the username logged in %u:

OIDCRemoteUserClaim preferred_username



Hamish

**Comment by bodtx** [comment]
Thank you indeed with
`%{OIDC_CLAIM_XXX}e`

where XXX is any key from id token, I cans log whatever I want.
